### PR TITLE
tanzu isolated-cluster should use system CA trust when not specified

### LINKED ISSUE #4173: isolated-cluster plugin does not follow CA trust conventions
**Bug description**

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (X) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

with the following:

```
  --source-ca-certificate
  --source-insecure
  --destination-ca-certificate string   The private repository’s CA certificate  (optional)
   --destination-insecure                Trusts the private repository’s certificate without validating it (optional)
```

the behaviour is counter to any downloader i have ever used. If the CA certificate isn't specified, then the system trust store isn't currently used. there's no file to point to for this on MacOS and Windows, which makes this unusable on those systems by any enterprise that wants to do airgapped and follow standard security practices. For Linux, the user would need to know the path to the system trust store which is different on different distributions.

I'm also not sure of the utility of the separate source-ca-certificate an destination-ca-certificate flags. Let's simply reduce to just `--ca-certificate` and `--insecure` and use the same set on either. By convention `--insecure` COULD have a `-k` as a shortflag to match other CLIs that copy curl semantics.

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by Vandy-P**
Signed-off-by: Vandana Pathak <pathakv@sjc-ubu-eng-127.vmware.com>

### What this PR does / why we need it

- tanzu isolated-cluster command is now using system CA trust when CA Cert is not specified. 
- Above functionality is internally provided by imgpkg module if external CA Cert is not provided, in this PR we remove external 
validation for CA Certificate.
- Other improvement in case of failure.
- Improve command help description. 
- Restructure Directory/package 
 

### Which issue(s) this PR fixes

Fixes #4173

### Describe testing done for PR
Tested tanzu isolated-cluster command without providing CA Cert .

### Release note
NA

### CODE REVIEW

**Review by randomvariable — CHANGES_REQUESTED** [review]
The help text still needs some work.

However, the logic does seem to be ok. I have only tested the download-bundle part.

File: cmd/cli/plugin/isolated-cluster/imagepullop/publishimagestotar.go [hunk 1]
```diff
@@ -37,8 +37,19 @@ type PublishImagesToTarOptions struct {
...
+	Short: "Download images/bundle from source repo into local disk (at current directory) as TAR files",
+	Example: `
+        # download images/bundle for TKG version v2.1.0 into local disk from repo projects.registry.vmware.com and authenticate source repo with default system CA certificate
+	tanzu isolated-cluster download-bundle --source-repo mirror-registry.test --tkg-version v2.1.0 
+
+	# download images/bundle for TKG version v2.1.0 into local disk from an internal registry without validating the certificate
+        tanzu isolated-cluster download-bundle --source-repo mirror-registry.test --tkg-version v2.1.0 --insecure
+
+	# download images/bundle for TKG version v2.1.0 into local disk from an internal mirror using a self-signed certificate
+        tanzu isolated-cluster download-bundle --source-repo mirror-registry.test --tkg-version v2.1.0 --ca-certificate registry.crt
```
Comment (randomvariable) [inline cmd/cli/plugin/isolated-cluster/imagepullop/publishimagestotar.go:50]:
```suggestion
	tanzu isolated-cluster download-bundle --source-repo mirror-registry.test/tkg --tkg-version v2.1.0 

	# download images/bundle for TKG version v2.1.0 into local disk from an internal registry without validating the certificate
        tanzu isolated-cluster download-bundle --source-repo mirror-registry.test/tkg --tkg-version v2.1.0 --insecure

	# download images/bundle for TKG version v2.1.0 into local disk from an internal mirror using a self-signed certificate
        tanzu isolated-cluster download-bundle --source-repo mirror-registry.test/tkg --tkg-version v2.1.0 --ca-certificate registry.crt
```

**Review by randomvariable — CHANGES_REQUESTED** [review]
Final changes please.

**Review by randomvariable — APPROVED** [review]

File: cmd/cli/plugin/isolated-cluster/README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ Usage:
 
 Examples:
     # copy image from projects.registry.vmware.com/tkg to /tmp folder
-    tanzu isolated-cluster download-bundle --source-repo projects.registry.vmware.com/tkg --tkg-version v1.6.0
+    tanzu isolated-cluster download-bundle --source-repo mirror-registry.test/tkg --tkg-version v1.6.0
```
Comment (jmoroski) [inline cmd/cli/plugin/isolated-cluster/README.md:26]: please update versions in this doc as well

**Review by jmoroski — APPROVED** [review]
Approved pending doc changes from previous comment

**Pull request merged**